src/ShellyRGBW2/shelly_init.cpp

```diff
@@ -43,13 +43,19 @@ void CreateComponents(std::vector<std::unique_ptr<Component>> *comps,
                       HAPAccessoryServerRef *svr) {
   auto *lb_cfg = (struct mgos_config_lb *) mgos_sys_config_get_lb1();
 
-  // if we are in mode 4 (RGBW) we have a white channel
-  Output *const out_w =
-      mgos_sys_config_get_shelly_mode() == 4 ? FindOutput(4) : nullptr;
+  std::unique_ptr<hap::RGBWLight> rgbw_light;
 
-  std::unique_ptr<hap::RGBWLight> rgbw_light(
-      new hap::RGBWLight(1, FindInput(1), FindOutput(1), FindOutput(2),
-                         FindOutput(3), out_w, lb_cfg));
+  if (mgos_sys_config_get_shelly_mode() == 4) {
+    rgbw_light.reset(new hap::RGBWLight(1, FindInput(1), FindOutput(1),
```

Comment:
reset isn't needed in that case, because there was no object assigned to it before
rgbw_light = new .........    should do the job